src/test/ui/const-generics/int-as-ref-const-param.stderr

```diff
@@ -0,0 +1,29 @@
+warning: the feature `const_generics` is incomplete and may cause the compiler to crash
+  --> $DIR/int-as-ref-const-param.rs:4:12
+   |
+LL | #![feature(const_generics, const_compare_raw_pointers)]
+   |            ^^^^^^^^^^^^^^
+   |
+   = note: `#[warn(incomplete_features)]` on by default
+
+error[E0308]: mismatched types
+  --> $DIR/int-as-ref-const-param.rs:23:25
+   |
+LL |     let _: Const<{A}> = Const::<{B}>;
+   |                         ^^^^^^^^^^^^ expected `$PREFIX0000000a : &()`, found `$PREFIX0000000b : &()`
+   |
+   = note: expected type `Const<$PREFIX0000000a : &()>`
+              found type `Const<$PREFIX0000000b : &()>`
+
+error[E0308]: mismatched types
+  --> $DIR/int-as-ref-const-param.rs:24:25
+   |
+LL |     let _: Const<{A}> = Const::<{&()}>;
+   |                         ^^^^^^^^^^^^^^ expected `$PREFIX0000000a : &()`, found `{pointer} : &()`
+   |
+   = note: expected type `Const<$PREFIX0000000a : &()>`
+              found type `Const<{pointer} : &()>`
```

Comment:
FWIW, on current nightly this prints
```
   = note: expected type `Const<Scalar(0x000000000000000a) : &()>`
              found type `Const<Scalar(AllocId(9).0x0) : &()>`
```
So, yeah, the new output is much better. :D

It seems like right now the code is shared between printing this error and printing the mir-dump, and that is likely not what we want?